front-end/src/util/__tests__/validation.test.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright 2019-2020 @paritytech/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { email, fullname, password, username } from '../validation';
+
+test('Email validation', () => {
+	const email1 = 'polk@dot.com';
+	const email2 = 'polk@@dot.com';
+	const email3 = 'polk@dot..com';
+
+	expect(email.pattern.test(email1)).toBe(true);
+	expect(email.pattern.test(email2)).toBe(false);
+	expect(email.pattern.test(email3)).toBe(false);
+});
+
+test('Fullname validation', () => {
+	const fullname1 = 'Tim';
+	const fullname2 = 'Tim   Cook';
+	const fullname3 = 'X Æ A-12';
+
+	expect(fullname1.length).toBeGreaterThanOrEqual(fullname.minLength);
+	expect(fullname2.length).toBeGreaterThanOrEqual(fullname.minLength);
+	expect(fullname3.length).toBeLessThanOrEqual(fullname.maxLength);
+});
+
+test('Password validation', () => {
+	const password1 = '123456';
+	const password2 = 'trailingSpace ';
+
+	expect(password1.length).toBeGreaterThanOrEqual(password.minLength);
```

Comment:
Same here, why testing the length? We should test the validation, e.g what was done in test 1
it('should fail with a short password') -> test with "123"
it('should fail with a empty password') -> test with ""
it('should pass with a complex password') -> test with "^qr09@}= £$/"
...

Reply:
Yeah, I get that.. But this is how `password` is defined in `validation.ts`
`export const password = { minLength: 6, required: true };`

There is no pattern property unlike with `email`

Reply:
ok then it doesn't make sense to test this. What needs to be tested (with enzyme) is the login component (which embeds this validation)